packages/ui/src/proposals/components/ProposalStages.tsx

```diff
@@ -15,31 +16,48 @@ export interface ProposalStagesProps extends ControlProps<number> {
 export const ProposalStages = ({ status, updates, constitutionality = '-', value, onChange }: ProposalStagesProps) => {
   const rounds = useMemo(() => {
     const decidingCount = updates.filter(({ status }) => status === 'deciding').length
+
+    const totalRoundCount = typeof constitutionality === 'string' ? undefined : constitutionality
+
     const lastUpdate = last(updates).status
     const onGoing = lastUpdate === status
     const approved = lastUpdate === 'gracing'
     const rejected = !onGoing && !approved
     const isDeciding = onGoing && status === 'deciding'
     const isDormant = onGoing && status === 'dormant'
+
     return [
       ...repeat((round) => ({ icon: <CheckboxIcon />, onClick: () => onChange(round) }), decidingCount - 1),
       {
         icon: !isDeciding && (rejected ? <CrossIcon /> : <CheckboxIcon />),
         onClick: () => onChange(decidingCount - 1),
       },
-      ...(isDormant ? [{ icon: false, onClick: undefined }] : []),
+      ...(isDormant
+        ? repeat(
+            () => ({ icon: false, onClick: undefined }),
+            totalRoundCount !== undefined ? totalRoundCount - decidingCount - 1 : 1
+          )
+        : []),
     ]
-  }, [updates.length, status])
+  }, [updates.length, status, constitutionality])
 
   return (
     <TabsContainer>
       {rounds.map(({ icon, onClick }, round) => (
-        <TabContainer key={round} active={round === value} disabled={!onClick} onClick={onClick}>
-          {icon}
-          <span>
-            Council approvals {round + 1}/{constitutionality}
-          </span>
-        </TabContainer>
+        <Tooltip
+          tooltipText={
+            round !== value
```

Comment:
Here the "disabled" tabs are those with `typeof onClick === 'undefined'`... Which is not very readable code.

IMO the best would be to change the type of `rounds`:
from: `{ icon: boolean, onClick: undefined | () => void }` to: `'approved' | 'rejected' | 'deciding' | 'disabled'`.
So this would become something like:
```tsx
{rounds.map((status, round) => {
  const isDisabled = status === 'disabled'
  const isActive = round === value
  const onClick = isDisabled ? undefined : () => onChange(round)
  const icon = iconMap[status]

  return (
  <Tooltip key={round} tooltipText={isDisabled ? "text1..." : "text2..."}>
    <TabContainer active={isActive} disabled={isDisabled} onClick={onClick}>
      {icon}
      <spann>
        ...
```

With `iconMap` defined on the module scope as:
```ts
const iconMap = { approved: <CheckboxIcon />, rejected: <CrossIcon />, deciding: undefined, disabled: undefined }
```
or even as a function with a switch...

Reply:
The [type of `status` is `ProposalStatus`](https://github.com/Joystream/pioneer/blob/6851ba0f445c676586eea8260017142ddc317e5c/packages/ui/src/proposals/components/ProposalStages.tsx#L11) defined [here](https://github.com/Joystream/pioneer/blob/dev/packages/ui/src/proposals/types/proposals.ts#L14) as `String`.

Reply:
@chrlschwb You're right about it currently being an object but this why I suggested to change it to: `'approved' | 'rejected' | 'deciding' | 'disabled'`. I guess calling the variable `status` is confusing maybe `roundStatus` would be better.